src/WebJobs.Script/Description/CSharp/CSharpFunctionInvoker.cs

```diff
@@ -0,0 +1,373 @@
+
+                await ProcessInputBindingsAsync(binder, executionContext, bindingData);
+
+                MethodInfo function = GetFunctionTarget();
+
+                var functionArguments = function.GetParameters();
+
+                object[] arguments = functionArguments
+                    .Select(p => p.IsOut ? null : parameters.FirstOrDefault(param => p.ParameterType.IsAssignableFrom(param.GetType())))
+                    .ToArray();
+
+                object functionResult = function.Invoke(null, arguments);
+
+                if (functionResult is Task)
+                {
+                    await ((Task)functionResult)
+                        .ContinueWith(t => functionResult = GetTaskResult(t));
+                }
+
+                IDictionary<string, object> functionOutputs = functionResult as IDictionary<string, object>;
+
+                // If we have a single output binding, directly specify its value
+                if (functionResult != null && _outputBindings.Count == 1)
+                {
+                    var outputBinding = _outputBindings.Single();
+
+                    if (functionOutputs == null || !functionOutputs.ContainsKey(outputBinding.Name))
+                    {
+                        functionOutputs = new Dictionary<string, object>
+                        {
+                            {outputBinding.Name, functionResult }
+                        };
+                    }
+                }
+                else if (functionOutputs == null)
+                {
+                    var output = functionArguments.Where(p => p.IsOut).ToList();
+                    functionOutputs = output.ToDictionary(o => o.Name, o => arguments[o.Position]);
+                }
+
+                await ProcessOutputBindingsAsync(_outputBindings, input, binder, bindingData, functionOutputs);
+
+                _fileTraceWriter.Verbose(string.Format("Function completed (Success)"));
+            }
+            catch (CompilationErrorException ex)
+            {
+                _fileTraceWriter.Error("Function compilation error");
+
+                foreach (var diagnostic in ex.Diagnostics.Where(d => !d.IsSuppressed))
+                {
+                    TraceLevel level = GetTraceLevelFromDiagnostic(diagnostic);
+                    _fileTraceWriter.Trace(new TraceEvent(level, diagnostic.ToString()));
+                }
+
+                _fileTraceWriter.Verbose(string.Format("Function completed (Failure)"));
+
+                throw;
+            }
+            catch (Exception ex)
+            {
+                _fileTraceWriter.Error(ex.Message, ex);
+                _fileTraceWriter.Verbose(string.Format("Function completed (Failure)"));
+                throw;
+            }
+        }
+
+        private MethodInfo GetFunctionTarget()
+        {
+            if (_function == null)
+            {
+                Script<object> script = CreateScript();
```

Comment:
This gets the function entry point. Additional files may be dynamically pulled in using the "#load "something.csx" directive.